eox_tagging/api/v1/viewset.py

```diff
@@ -30,25 +30,52 @@ def get_queryset(self):
         """Restricts the returned tags."""
         owner_type = self.request.query_params.get("owner_type")
         include_invalid = self.request.query_params.get("include_invalid")
-        user = self.request.user
-        site = get_site()
 
         if include_invalid and include_invalid.lower() in ["true", "1"]:
             queryset = Tag.objects.all()
         else:
             queryset = Tag.objects.valid()
 
-        if owner_type:
-            owner_id = {"username": user.username} if owner_type == "user" else {"id": site.id}
-            try:
-                queryset = queryset.find_by_owner(owner_type=owner_type, owner_id=owner_id)
-                return queryset
-            except Exception:  # pylint: disable=broad-except
-                return queryset.none()
+        owner_information = self.__get_request_owner(owner_type)
 
         try:
-            queryset = queryset.find_by_owner(owner_type="site", owner_id={"id": site.id}) \
-                | queryset.find_by_owner(owner_type="user", owner_id={"username": user.username})
-            return queryset
+            queryset_union = queryset.none()
+            for owner in owner_information:
+                queryset_union |= queryset.find_by_owner(**owner)
+
+            return queryset_union
         except Exception:  # pylint: disable=broad-except
             return queryset.none()
+
+    def __get_request_owner(self, owner_type):
+        """Returns the owners of the tag to filter the queryset."""
+        site = self.__get_site()
+        user = self.__get_user()
+
+        if owner_type is None:
+            return [site, user]
+
+        if owner_type.lower() == "user":
+            return user
+        elif owner_type.lower() == "site":
```

Comment:
```suggestion
        if owner_type.lower() == "site":
```

add blank space before this line